src/System/Diagnostic/SourceCodeIntegrityChecker.php

```diff
@@ -0,0 +1,248 @@
+    }
+
+    private function getGLPIRelease(&$errors = []): string|null
+    {
+        $gh_releases_endpoint = 'https://api.github.com/repos/glpi-project/glpi/releases';
+        $version_to_get = VersionParser::getNormalizedVersion(GLPI_VERSION);
+        $client = new Client();
+        $dest = null;
+        try {
+            $response = $client->request('GET', $gh_releases_endpoint);
+            $releases = json_decode((string) $response->getBody(), true, 512, JSON_THROW_ON_ERROR);
+            foreach ($releases as $release) {
+                if ($release['tag_name'] === $version_to_get) {
+                    foreach ($release['assets'] as $asset) {
+                        if (str_starts_with($asset['name'], 'glpi-' . $version_to_get)) {
+                            $dest = GLPI_TMP_DIR . '/' . $asset['name'];
+                            // Check if the file already exists in the tmp dir
+                            if (file_exists($dest)) {
+                                break;
+                            }
+                            $url = $asset['browser_download_url'];
+                            $client->request('GET', $url, ['sink' => $dest]);
+                            break;
+                        }
+                    }
+                    break;
+                }
+            }
+        } catch (GuzzleException $e) {
+            $errors[] = $e->getMessage();
+        }
+        return $dest;
+    }
+
+    public function getDiff(array &$errors = [], bool $allow_download = false): string
+    {
+        $summary = $this->getSummary();
+        // ignore OK files in case they are present
+        $summary = array_filter($summary, static fn ($status) => $status !== self::STATUS_OK);
+        // Ensure the release is downloaded
+        if ($allow_download) {
+            $release_path = $this->getGLPIRelease($errors);
+        } else {
+            $release_path = GLPI_TMP_DIR . '/glpi-' . VersionParser::getNormalizedVersion(GLPI_VERSION) . '.tgz';
+            if (!file_exists($release_path)) {
+                $errors[] = 'The release is not downloaded and downloading it was not allowed';
+                return '';
+            }
+        }
+        if ($release_path === null) {
+            return 'An error occurred while downloading the release';
+        }
+        $diff = '';
+
+        foreach ($summary as $file => $status) {
+            // print diff --git line for the file
+            $diff .= 'diff --git a/' . $file . ' b/' . $file . "\n";
+            $original_file = 'a/' . $file;
+            $current_file = 'b/' . $file;
+            $original_content = '';
+            $current_content = '';
+            // If status is missing, just add the info to the diff string
+            if ($status === self::STATUS_MISSING) {
+                $current_file = '/dev/null';
+            } else if ($status === self::STATUS_ADDED || !file_exists('phar://' . $release_path . '/' . $file)) {
```

Comment:
In the official release archive, all files are under a `glpi` directory.
```suggestion
            } else if ($status === self::STATUS_ADDED || !file_exists('phar://' . $release_path . '/glpi/' . $file)) {
```